factory/animal.h

```diff
@@ -0,0 +1,54 @@
+
+#ifndef __ANIMAL_PLANET_ANIMAL_HPP_
+#define __ANIMAL_PLANET_ANIMAL_HPP_
+
+#include <memory>
+#include <string>
+
+namespace animal
+{
+
+//using AnimalPtr = std::unique_ptr<CAnimal>;
+class CAnimal
+{
+public:
+    CAnimal(std::string sName = {}, std::string sType = "");
+
+    virtual std::string GetType() const = 0;
```

Comment:
Why these methods are pure virtual?
Do we need to override them separately for each derived class?